### PR TITLE
Jump fixes

### LINKED ISSUE #1936: Skip jump in Event Player
**Name of the game**: Super Crystal Hunter
**Player platform**: Player 0.6.1, Win10 x64
**Describe the issue in detail and how to reproduce it**:

Create an event as Player Touch, then introduce the next commands:
```
@> Set Move Route: Player,
$>Direction Fix ON
$>Through ON
$>Turn Left
$>Graphic: (center charset frame, in my case left graphic)
$>Wait
$>Wait
$>Jump
$>Turn right
$>Graphic: (another center charset frame)
$> Move Upper Right
$>Wait
$>Wait
$>Land
$>Turn left
$>Graphic: (another center charset frame)
$>Wait
$>Wait
$>Turn right
$>Graphic: (center charset frame, in my case left graphic)
$>Through OFF
$>Direction Fix OFF
@> Wait for All Movement
```
Basically it's a jump from the floor to a box and the change graphics are when the event Player prepare its jump (kneel, and change to normal frames)

### LINKED ISSUE #1937: Possible z-order event bug
**Name of the game**: Super Crystal Hunter
**Player platform**: Player 0.6.1, Win10 x64
**Describe the issue in detail and how to reproduce it**:

Create 2 events.

One of them with the next commands in parallel process:

`
@> Set Move Route: [Jumper], Jump , Land
@> Wait: 0.1 seconds
`

The 2nd event will be the "Jumper" event, configured with:

- Priority: Same as character
- Animation Type: Fixed Graphic (in pattern frame- Right)

The video shows the error in Player 0.6.1
[z-index _error.zip](https://github.com/EasyRPG/Player/files/3831367/z-index._error.zip)

### LINKED ISSUE #1938: Show Animation follows event
**Name of the game**: Super Crystal Hunter
**Player platform**: Player 0.6.1, Win10 x64
**Describe the issue in detail and how to reproduce it**:

Create 2 events.
Event A in Parallel Process with:

`@> Set Move Route: [EventB], Wait, Jump , Land , Wait @> Wait: 0.5 seconds`

EventB: in Fixed graphic with pattern left and its priority as "same as character".
In its contents, put any command "Show animation" and interact with event Player.

Video showing all of that in Player 0.6.1:
[animation-follow-event.zip](https://github.com/EasyRPG/Player/files/3831430/animation-follow-event.zip)

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix #1938 
Fix #1937 
Fix #1936 

@Br4ssman  Please confirm fixes in your game

### CODE REVIEW

**Comment by Br4ssman** [comment]
It working perfectly.
You rocks!

File: src/game_character.h [hunk 1]
```diff
@@ -649,9 +649,10 @@ class Game_Character {
 	 * Gets sprite y coordinate transformed to screen coordinate in pixels.
 	 *
 	 * @param apply_shift When true the coordinate is shifted by the map height (for looping maps)
+	 * @param apply_jump Apply jump height modifier if character is jumping
 	 * @return screen y coordinate in pixels.
 	 */
-	virtual int GetScreenY(bool apply_shift = false) const;
+	virtual int GetScreenY(bool apply_shift = false, bool apply_jump=true) const;
```
Comment (carstene1ns) [inline src/game_character.h]: spaces?

File: src/game_vehicle.h [hunk 2]
```diff
@@ -67,7 +67,7 @@ class Game_Vehicle : public Game_Character {
 	bool IsAboard() const;
 	void SyncWithPlayer();
 	void AnimateAscentDescent();
-	int GetScreenY(bool apply_shift = false) const override;
+	int GetScreenY(bool apply_shift = false, bool apply_jump=true) const override;
```
Comment (carstene1ns) [inline src/game_vehicle.h]: same.

**Comment by fmatthew5876** [comment]
Fixed white space issues and rebased

**Pull request merged**